src/components/AttachmentPicker/index.native.js

```diff
@@ -250,136 +148,207 @@ class AttachmentPicker extends Component {
+     *
+     * @param {Array<ImagePickerResponse|DocumentPickerResponse>} attachments
+     * @returns {Promise}
+     */
+    const pickAttachment = useCallback((attachments = []) => {
+        if (attachments.length === 0) {
+            return;
+        }
+
+        const fileData = _.first(attachments);
+
+        if (fileData.width === -1 || fileData.height === -1) {
+            showImageCorruptionAlert();
+            return;
+        }
+
+        return getDataForUpload(fileData)
+            .then((result) => {
+                completeAttachmentSelection.current(result);
+            })
+            .catch((error) => {
+                showGeneralAlert(error.message);
+                throw error;
+            });
+    }, [showGeneralAlert, showImageCorruptionAlert])
+
     /**
      * Setup native attachment selection to start after this popover closes
      *
      * @param {Object} item - an item from this.menuItemData
      * @param {Function} item.pickAttachment
      */
-    selectItem(item) {
+    const selectItem = useCallback((item) => {
         /* setTimeout delays execution to the frame after the modal closes
          * without this on iOS closing the modal closes the gallery/camera as well */
-        this.onModalHide = () =>
+        onModalHide.current = () =>
             setTimeout(
                 () =>
                     item
                         .pickAttachment()
-                        .then(this.pickAttachment)
+                        .then(pickAttachment)
                         .catch(console.error)
-                        .finally(() => delete this.onModalHide),
+                        .finally(() => delete onModalHide.current),
                 200,
             );
 
-        this.close();
+        close();
+    }, [pickAttachment])
+
+    const attachKeyboardListener = useCallback(() => {
+        const shortcutConfig = CONST.KEYBOARD_SHORTCUTS.ENTER;
+        keyboardListener.current = KeyboardShortcut.subscribe(
+            shortcutConfig.shortcutKey,
+            () => {
+                if (focusedIndex === -1) {
+                    return;
+                }
+                selectItem(menuItemData[focusedIndex]);
+                setFocusedIndex(-1); // Reset the focusedIndex on selecting any menu
+            },
+            shortcutConfig.descriptionKey,
+            shortcutConfig.modifiers,
+            true,
+        );
+    }, [focusedIndex, menuItemData, selectItem])
+
+    const removeKeyboardListener = () => {
+        if (!keyboardListener.current) {
+            return;
+        }
+        keyboardListener.current();
     }
 
+    useEffect(() => {
+        // When selecting an image on a native device, it would be redundant to have a second option for choosing a document,
+        // so it is excluded in this case.
+        if (type === CONST.ATTACHMENT_PICKER_TYPE.IMAGE) {
+            return;
+        }
+
+        setMenuItemData((oldMenuItemData) => [
```

Comment:
Can't this be moved to the useState initialisation? It would work the same and this useEffect could be discarded or compacted

Reply:
Converting it to a useMemo